src/main/java/com/emc/mongoose/base/load/step/client/LoadStepClientBase.java

```diff
@@ -394,7 +394,21 @@ public final boolean await(final long timeout, final TimeUnit timeUnit)
 							.map(
 											stepSlice -> {
 												try {
-													return stepSlice.await(timeout, timeUnit);
+													var invokeTimeMillis = System.currentTimeMillis();
+													var timeOutMillis = timeUnit.toMillis(timeout);
+													var await_result = false;
+													long elapsedTimeMillis;
+													while(timeOutMillis > (elapsedTimeMillis = System.currentTimeMillis() - invokeTimeMillis)) {
+														elapsedTimeMillis = System.currentTimeMillis() - invokeTimeMillis;
+														if (timeOutMillis <= elapsedTimeMillis) {
+															break;
+														}
+														if (Thread.currentThread().isInterrupted()){
+															throwUnchecked(new InterruptedException());
```

Comment:
Isn't it enough to `break` the loop?

Reply:
Isn’t it better to see a stacktrace, as is the case with stndalone mode?

Reply:
maybe